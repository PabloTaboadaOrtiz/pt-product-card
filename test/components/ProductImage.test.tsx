import * as React from 'react';
import renderer from 'react-test-renderer';
import { ProductImage, ProductCard } from '../../src/components';
import { PRODUCT_2 } from '../data/products';

describe('ProductImage', () => {
  test('debe de mostar el componente', () => {
    const wrapper = renderer.create(
      <ProductImage img="https://hola.jpg"></ProductImage>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  test('debe de mostar el componente con la imagen del producto', () => {
    const wrapper = renderer.create(
      <ProductCard product={PRODUCT_2}>
        {() => <ProductImage></ProductImage>}
      </ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
