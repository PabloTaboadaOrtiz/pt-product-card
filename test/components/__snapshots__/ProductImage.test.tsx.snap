// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage debe de mostar el componente 1`] = `
<img
  alt="product image"
  className="productImg undefined"
  src="https://hola.jpg"
/>
`;

exports[`ProductImage debe de mostar el componente con la imagen del producto 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="product image"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;
